### DISCUSSION TITLE
sloppiness not work

### DISCUSSION THREAD

**Discussion by MdShadabAhamad** [excalidraw/excalidraw#8579] [General]
const boothElements = booths.map((booth) => ({
          type: "rectangle",
          id: booth.locationid,
          version: 2,
          versionNonce: Date.now(),
          x: booth.location_x,
          y: booth.location_y,
          width: booth.location_width,
          height: booth.location_height,
          angle: 0,
          strokeColor: "#000000",
          backgroundColor: booth.color || "#cccccc",
          fillStyle: "solid",
          strokeWidth: 1,
          strokeStyle: "solid",
          roughness: 1,
          opacity: 100,
          groupIds: [],
          strokeSharpness: "sharp",
          seed: Date.now(),
          updated: Date.now(),
          locked: false,
          sloppiness: "Architect",
        }));

**Comment by Mrazator** [comment]
You will need to define "rougness" prop which is an integer. Refer to:

```
export const ROUGHNESS = {
  architect: 0,
  artist: 1,
  cartoonist: 2,
} as const;

export const STROKE_WIDTH = {
  thin: 1,
  bold: 2,
  extraBold: 4,
} as const;

export const DEFAULT_ELEMENT_PROPS: {
  strokeColor: ExcalidrawElement["strokeColor"];
  backgroundColor: ExcalidrawElement["backgroundColor"];
  fillStyle: ExcalidrawElement["fillStyle"];
  strokeWidth: ExcalidrawElement["strokeWidth"];
  strokeStyle: ExcalidrawElement["strokeStyle"];
  roughness: ExcalidrawElement["roughness"];
  opacity: ExcalidrawElement["opacity"];
  locked: ExcalidrawElement["locked"];
} = {
  strokeColor: COLOR_PALETTE.black,
  backgroundColor: COLOR_PALETTE.transparent,
  fillStyle: "solid",
  strokeWidth: 2,
  strokeStyle: "solid",
  roughness: ROUGHNESS.artist,
  opacity: 100,
  locked: false,
};
```

**Comment by MdShadabAhamad** [comment]
hi, thanks for your response , i also have another query that , how i used
library ,  when i click on used library in my local then it redirect to
blank page then redirect to my page only but  library is not added. I
searched for it but I did not get any solution .

**Reply by Mrazator** [reply]
Could be a bug, feel free to create an issue if you have exact repro steps.
